plugins/test/fqdn_valid.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2023, Vladimir Botka <vbotka@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+  name: fqdn_valid
+  short_description: Validates fully-qualified domain names against RFC 1123.
+  version_added: 8.0.3
+  author: Vladimir Botka (@vbotka)
+  requirements:
+  - fqdn>=1.5.1 (PyPI)
+  description:
+    - This test validates Fully Qualified Domain Names (FQDNs)
+      conforming to the Internet Engineering Task Force specification
+      RFC 1123 and RFC 952
+    - The design intent is to validate that a string would be
+      traditionally acceptable as a public Internet hostname to
+      RFC-conforming software, which is a strict subset of the logic
+      in modern web browsers like Mozilla Firefox and Chromium that
+      determines whether make a DNS lookup.
+    - Certificate Authorities like Let's Encrypt run a narrower set of
+      string validation logic to determine validity for issuance. This
+      test is not intended to achieve functional parity with CA
+      issuance
+    - Single label names are allowed by default (min_labels=1).
+  options:
+    name:
+      description: Name of the host.
+      type: str
+      required: true
+    min_labels:
+      description: Required minimum of labels, separated by period
+      default: 1
+      type: int
+      required: false
+    allow_underscores:
+      description: Allow underscore characters.
+      default: False
+      type: bool
+      required: false
+'''
+
+EXAMPLES = '''
+- name: Make sure that hostname is valid
+  ansible.builtin.assert:
+    that: hostname is community.general.fqdn_valid
+
+- name: Make sure that hostname is at least 3 labels long (a.b.c)
+  ansible.builtin.assert:
+    that: hostname is community.general.fqdn_valid(min_labels=3)
+
+- name: Make sure that hostname is at least 2 labels long (a.b). Allow '_'
+  ansible.builtin.assert:
+    that: hostname is community.general.fqdn_valid(min_labels=2, allow_underscores=True)
+'''
+
+RETURN = '''
+  _value:
+    description: Whether the name is valid.
+    type: bool
+'''
+
+from fqdn import FQDN
```

Comment:
This import needs to be guarded (see https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/import.html).